TATB - 42 : Add user profile [feature/development]

PrabhatJTDeveloper opened the pull request.

PrabhatJTDeveloper committed:
- TATB-37 redesign the quiz test template
  - for that change the html and scss file of the quiz component .

PrabhatJTDeveloper committed:
- TATB 37
   - removed unnecessary id and role in the button from the html file
   - removed unwanted css in scss file

PrabhatJTDeveloper committed:
- TATB 37
   - removed unwanted id and class from carousel in html file
   - removed the css which is also used as boostrap class.

PrabhatJTDeveloper committed:
- TATB 37
  - conflicts occurs in quiz.component html and scss file so solve it
  - change card.value with card.id in quiz component html file

PrabhatJTDeveloper committed:
- TATB 37 removed unused div in html file

PrabhatJTDeveloper committed:
- TATB 37
  - make change in condition of hightlighted text in quiz.component.html file

PrabhatJTDeveloper committed:
- TATB 37
   - Solved the conflicts in quiz component html and css file.

PrabhatJTDeveloper committed:
- TATB 37
   - give the boostrap class and remove from the css
   - and make change in html file

PrabhatJTDeveloper committed:
- TATB 37
   - make changes in html and scss file.

PrabhatJTDeveloper committed:
- TATB 37
   - removed "this" from the {{this.selectedQuizType}} in the quiz component html file.

PrabhatJTDeveloper committed:
- TATB 37
  - removed "this" from the html file

PrabhatJTDeveloper committed:
- TATB 37
   - make quiz test template responsive till mobile view.
   - also make image responsive in instruction page.

PrabhatJTDeveloper committed:
- solve the conflicts in the quiz component html and scss file.

PrabhatJTDeveloper committed:
- remove fixed height of header and footer.

PrabhatJTDeveloper committed:
- TATB 37
   - solve the conflicts on quiz component

PrabhatJTDeveloper committed:
- solve the conflits.

PrabhatJTDeveloper committed:
- TATB 42
  - design the user profile page.
  - mange data from localstorage.

PrabhatJTDeveloper committed:
- solve the conflict

PrabhatJTDeveloper committed:
- TATB 42
  - make edit and validation for user profile .
  - added user profile in header dropdown

PoojanPatelJT commented on src/app/component/quiz/quiz.component.html:
```diff
@@ -19,21 +19,16 @@
                     <div
-                      class="card-header header-txt py-0 px-3 d-flex align-items-center justify-content-between position-relative"
+                      class="card-header header-txt py-3 px-3 d-flex align-items-center justify-content-between position-relative"
                     >
```
instead of giving px-3 and py-3 individually replace it with p-3

PrabhatJTDeveloper replied:
ok, i changed it.

PoojanPatelJT commented on src/app/component/quiz/quiz.component.html:
```diff
@@ -93,11 +87,11 @@
                     <div
-                      class="card-footer footer-txt py-0 px-3 d-flex align-items-center justify-content-between position-relative"
+                      class="card-footer footer-txt py-3 px-3 d-flex align-items-center justify-content-between position-relative"
                     >
```
same here

PrabhatJTDeveloper replied:
yes, i changed it.

PoojanPatelJT commented on src/app/component/user-profile/user-profile.component.html:
```diff
@@ -0,0 +1,118 @@
+                     <div class="form-text text-danger fixed-height">
+                      <span
+                        *ngIf="profilePageFormController['dateOfBirth'].errors?.['required'] && (profilePageFormController['dateOfBirth'].touched || profilePageFormController['dateOfBirth'].dirty)"
+                      >
```
remove commentted code

PrabhatJTDeveloper replied:
this code is usable so i have updated it.

PoojanPatelJT commented on src/app/store/autentication/autentication.action.ts:
```diff
@@ -8,6 +8,8 @@ export const validateSessionSucess = createAction('[Autentication] validateSessi
 export const loadUserSuccess = createAction('[Autentication] loadUserSuccess',props<{users:any}>());
+export const updateUserDetails = createAction('[Autentication] updateUserDetails',props<{users:any}>());
+export const updateUserDetailsSucess = createAction('[Autentication] updateUserDetailsSucess',props<{users:any}>());
 
```
add proper data type

PrabhatJTDeveloper replied:
ok, i have done it.

PoojanPatelJT commented on src/app/store/autentication/autentication.effect.ts:
```diff
@@ -46,6 +46,23 @@ export class AuthEffects implements OnDestroy {
+      switchMap((payload) => {
+        return this.authSerivce.updateUserDetails(payload).pipe(
+          map((users: any) => autenticationAction.updateUserDetailsSucess({users})),
+          catchError((error: any) =>
```
here also add proper data type

PrabhatJTDeveloper replied:
yes, i have done it.

PrabhatJTDeveloper committed:
Merge branch 'feature/development' of https://github.com/Jyoti-Technosoft/technical-assistant into feature/TATB-42-add-user-profile

PrabhatJTDeveloper committed:
TAB:42
 -implemented logic for update detials with password field

PrabhatJTDeveloper committed:
-- deleted form on reset field

PrabhatJTDeveloper committed:
- implement payload for password field.

PrabhatJTDeveloper committed:
- TATB 42
  - cahnge boostrap class in quiz component
  - cahnge in button color and hover color
  - change the declaration of letters and name printing

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+  takeUntil,
+} from 'rxjs';
+import { Store } from '@ngrx/store';
+import { Router } from '@angular/router';
```
please add space herer

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+import { AuthenticationService } from '@app/service/authentication.service';
+import { autenticationState } from '@app/store/autentication/autentication.state';
+import {
+  AbstractControl,
```
also this will not in this position

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+  Validators,
+} from '@angular/forms';
+import { updateUserDetails } from '@app/store/autentication/autentication.action';
+import { NgbCalendar } from '@ng-bootstrap/ng-bootstrap';
```
do proper code intedation

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+@Component({
+  selector: 'app-user-profile',
+  templateUrl: './user-profile.component.html',
+  styleUrls: ['./user-profile.component.scss'],
```
remove extra comma

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+  styleUrls: ['./user-profile.component.scss'],
+})
+export class UserProfileComponent implements OnInit, AfterViewInit {
+  loggedInUser$: Observable<any> | undefined;
```
please provide here type nay will not work

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+  loggedInUser$: Observable<any> | undefined;
+  destroyer$: ReplaySubject<boolean> = new ReplaySubject();
+  profilePageForm!: FormGroup;
+  editMode: boolean = false;
```
why two diffrent variable for same thing ?

charmijtdev commented on src/app/component/user-profile/user-profile.component.ts:
```diff
@@ -0,0 +1,202 @@
+  }
+
+  updateUserDetails() {
+    console.log(this.profilePageForm?.value);
```
please remove console from here

PrabhatJTDeveloper committed:
- TATB 42
    - removed unwanted comma.
    - done code intedation.

PrabhatJTDeveloper committed:
- TATB 42
  - changed the password error printing
  - put the icon on buttons when it is in mobile view.
